src/cdk/testing/protractor/protractor-element.ts

```diff
@@ -129,8 +129,14 @@ export class ProtractorElement implements TestElement {
     return this.element.sendKeys(...keys);
   }
 
-  async text(): Promise<string> {
-    return this.element.getText();
+  async text(options?: {excludes?: string}): Promise<string> {
+    if (options?.excludes) {
+      return browser.executeScript(`var clone = arguments[0].cloneNode(true); 
```

Comment:
I _think_ you might be able to make a shared function between the unit and e2e tests called e.g.
```
function getTextWithExcludedElements(element, excludeSelector) {...}
```
and then do
```
browser.executeScript(getTextWithExcludedElement, element, excludeSelector)
```
This will boil down to the same thing, it'll just `toString` the function and pass it along, but at least it'll be easier to read in the IDE. If that doesn't work, at least indent like this:
```
return browser.executeScript(`
  var clone = arguments[0].cloneNode(true);
  var remove = clone.querySelectorAll(arguments[1]);
  remove.forEach(n => n.remove());
  return clone.textContent
`, this.element, options.excludes);
```